### PR TITLE
Cleanup the build

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR attemps to clean up the build as follows:

- build python bindings independently against pre-built libraries, rather than recompiling dependencies over and over
- don't build daemons as much as possible, since they don't get packaged in the end

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@jasoncpatton / @timtheisen, are you able to give these changes a once-over?

File: recipe/condor_config.generic.conda.patch [hunk 1]
```diff
@@ -0,0 +1,59 @@
+--- a/src/condor_examples/condor_config.generic
++++ b/src/condor_examples/condor_config.patched
```
Comment (jasoncpatton) [inline recipe/condor_config.generic.conda.patch:2]: Maybe (in the future) worth contributing to https://github.com/htcondor/htcondor/tree/master/src/condor_examples and instead patching from upstream?

**Review by jasoncpatton — APPROVED** [review]
I'm not exactly sure what the result of all of the cmake cruft cleanup is just looking at this, but if it's working for you, it should be fine (and good... there's a lot of cruft).

File: recipe/condor_config.generic.conda.patch (hunk 1 above)
Comment (duncanmmacleod) [reply recipe/condor_config.generic.conda.patch:2]: I can consider that, thanks @jasoncpatton.

**Comment by duncanmmacleod** [comment]
> I'm not exactly sure what the result of all of the cmake cruft cleanup is just looking at this, but if it's working for you, it should be fine (and good... there's a lot of cruft).

The cmake patches just remove build hooks for as many of the daemons as possible - we don't distribute them in the end, so if we can speed up the build by not building them either, all the better.

It would be great to have a cmake flag to disable daemon building...

**Pull request merged**